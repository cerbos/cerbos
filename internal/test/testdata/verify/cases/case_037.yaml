description: "Output matching failure (#1455)"
