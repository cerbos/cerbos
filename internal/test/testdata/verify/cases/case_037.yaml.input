-- example.yaml --
---
apiVersion: api.cerbos.dev/v1
resourcePolicy:
  resource: example
  version: default
  rules:
    - actions:
        - view
      effect: EFFECT_ALLOW
      roles:
        - user
      condition:
        match:
          expr: `request.principal.attr.ip == "10.20.2.2"`
      output:
        when:
          ruleActivated: R

-- example_test.yaml --
---
name: Example test suite
principals:
  exampleUser:
    id: example.user@yourapp.example.com
    roles:
      - user
    attr:
      ip: 10.20.2.2
resources:
  exampleResource:
    kind: example
    id: example-001
    attr:
      owner: example.user@yourapp.example.com
tests:
  - name: Example test
    input:
      principals:
        - exampleUser
      resources:
        - exampleResource
      actions:
        - view
    expected:
      - principal: exampleUser
        resource: exampleResource
        actions:
          view: EFFECT_ALLOW
        outputs:
          - action: edit
            expected:
              - src: x
                val: x
