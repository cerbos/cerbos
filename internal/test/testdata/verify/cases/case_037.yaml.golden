{
  "suites": [
    {
      "file": "example_test.yaml",
      "name": "Example test suite",
      "summary": {
        "overallResult": "RESULT_ERRORED"
      },
      "error": "Failed to load the test suite: invalid test \"Example test\": found output expectations for actions that are not in the input actions list: [edit]"
    }
  ],
  "summary": {
    "overallResult": "RESULT_ERRORED"
  }
}
